options.go

```diff
@@ -9,9 +9,18 @@ import (
 	"srcd.works/go-git.v4/plumbing/transport"
 )
 
+// SubmoduleRescursivity defines how depth will affect any submodule recursive
+// operation
+type SubmoduleRescursivity int
```

Comment:
alternative name: `SubmoduleRecursionDepth`, explain what negative values will do and how to make it infinite.